Lazy/Lazy/MultiThreadLazy.cs

```diff
@@ -0,0 +1,49 @@
+// <copyright file="MultiThreadLazy.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace Lazy;
+/// <summary>
+/// A lazy evaluation implementation for multithreaded environments.
+/// Ensures that the value is computed only once, even in concurrent access scenarios.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public class MultiThreadLazy<T>: ILazy<T>
+{
+    private Func<T>? _supplier;
+    private T _value = default;
+    private bool _isCalculated = false;
+    private readonly object _lockObject = new object();
```

Comment:
```suggestion
    private readonly object _lockObject = new();
```
Не пишите один тип дважды